agent/generators.py

```diff
@@ -43,7 +43,9 @@ def get_services(
                 ports = [ports]
             for port in ports:
                 data["port"] = int(port.get("@portid"))
-                data["protocol"] = port.get("@protocol")
+                data["protocol"] = port.get("service", {}).get(
```

Comment:
What is the purpose of having data["protocol"] and data["service"] with the same value?